flake.nix

```diff
@@ -15,7 +15,7 @@
     };
 
     nix = {
-      url = "https://flakehub.com/f/NixOS/nix/=2.20.5.tar.gz";
+      url = "https://flakehub.com/f/DeterminateSystems/nix/2.0.0";
```

Comment:
We should pin to exact versions, so I don't have to look in the flake.lock to see what version of Nix `2.0.0` refers to at the current point in time.

It would also be a good idea to make this upstream automatically notice new Nix versions and make them available (even if we don't set them to be the version we serve from `install.det.sys`), just like `NixOS/nix` does.